src/EnergyPlus/WaterManager.cc

```diff
@@ -1627,6 +1690,49 @@ namespace WaterManager {
         }
     }
 
+    void ReportRainfall(EnergyPlusData &state)
+    {
+        Array1D_string const Months(12, {"Jan", "Feb", "Mar", "Apr", "May", "Jun", "Jul", "Aug", "Sep", "Oct", "Nov", "Dec"});
+        for (int i = 0; i < 12; i++) {
+            OutputReportPredefined::PreDefTableEntry(state,
+                                                     state.dataOutRptPredefined->pdchMonthlyTotalPrecInWeather,
+                                                     Months[i],
+                                                     state.dataWaterData->RainFall.MonthlyTotalPrecInWeather[i]);
+            OutputReportPredefined::PreDefTableEntry(
+                state, state.dataOutRptPredefined->pdchMonthlyTotalHrRain, Months[i], state.dataWaterData->RainFall.numRainyHoursInWeather[i]);
+        }
+        if (state.dataWaterData->NumWaterStorageTanks > 0) {
+            for (int i = 0; i < 12; i++) {
+                OutputReportPredefined::PreDefTableEntry(state,
+                                                         state.dataOutRptPredefined->pdchMonthlyTotalPrecInRainCol,
+                                                         Months[i],
+                                                         state.dataWaterData->RainFall.MonthlyTotalPrecInRainCol[i]);
+                Real64 accVolCollectedMonthly = 0.0;
+                for (int Item = 1; Item <= state.dataWaterData->NumRainCollectors; Item++) {
+                    accVolCollectedMonthly += state.dataWaterData->RainCollector(Item).VolCollectedMonthly[i];
+                }
+                OutputReportPredefined::PreDefTableEntry(
+                    state, state.dataOutRptPredefined->pdchMonthlyTotalRainCol, Months[i], accVolCollectedMonthly);
+            }
+        }
+        // check whether there are eco roofs
+        bool ecoroofFlag = false;
+        for (int ConstrNum = 1; ConstrNum <= state.dataHeatBal->TotConstructs; ++ConstrNum) {
+            ecoroofFlag = (state.dataConstruction->Construct(ConstrNum).TypeIsEcoRoof) || ecoroofFlag;
+        }
+        // report ecoroof
+        if (ecoroofFlag) {
+            for (int i = 0; i < 12; i++) {
+                OutputReportPredefined::PreDefTableEntry(state,
+                                                         state.dataOutRptPredefined->pdchMonthlyTotalPrecInRoofIrr,
+                                                         Months[i],
+                                                         state.dataWaterData->RainFall.MonthlyTotalPrecInRoofIrr[i]);
+                OutputReportPredefined::PreDefTableEntry(
+                    state, state.dataOutRptPredefined->pdchMonthlyTotalIrrDep, Months[i], state.dataEcoRoofMgr->MonthlyIrrigation[i]);
+            }
+        }
```

Comment:
Seems like this variable initialization, loop to set the variable, then check the variable could be reduced to just:
```
for (auto const & c : state.dataHeatBal->TotConstructs) {
  if (c.TypeIsEcoRoof) {
    for (int i = 0; i < 12, i++) {
      OutputReportPredefined::PreDefTableEntry( ... blah ... );
      OutputReportPredefined::PreDefTableEntry( ... blah ... );
    }
    break;
  }
}
````

Reply:
Good point. I change the loop logit to this pattern.